src/System.Composition.Convention/src/Microsoft/Internal/Lock.Reader.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+
+namespace Microsoft.Internal
+{
+    internal struct ReadLock : IDisposable
+    {
+        private readonly Lock _lock;
+        private int _isDisposed;
+
+        public ReadLock(Lock @lock)
+        {
+            _isDisposed = 0;
+            _lock = @lock;
+            _lock.EnterReadLock();
+        }
+
+        public void Dispose()
+        {
+            if (Interlocked.CompareExchange(ref _isDisposed, 1, 0) == 0)
```

Comment:
Cool, I'd guess those are the only ones.

Thinking a bit more about how this is used (during reflection, which is done once only for each component type IIRC), I'd say the risk of any major regression making the change is pretty low.